doc/reference/storage_lvm.md

```diff
@@ -1,45 +1,62 @@
 (storage-lvm)=
 # LVM - `lvm`
 
- - Uses LVs for images, then LV snapshots for instances and instance snapshots.
- - The filesystem used for the LVs is ext4 (can be configured to use xfs instead).
- - By default, all LVM storage pools use an LVM thinpool in which logical
-   volumes for all LXD storage entities (images, instances, etc.) are created.
-   This behavior can be changed by setting "lvm.use\_thinpool" to "false". In
-   this case, LXD will use normal logical volumes for all non-instance
-   snapshot storage entities (images, instances, etc.). This means most storage
-   operations will need to fallback to rsyncing since non-thinpool logical
-   volumes do not support snapshots of snapshots. Note that this entails
-   serious performance impacts for the LVM driver causing it to be close to the
-   fallback DIR driver both in speed and storage usage. This option should only
-   be chosen if the use-case renders it necessary.
- - For environments with high instance turn over (e.g continuous integration)
-   it may be important to tweak the archival `retain_min` and `retain_days`
-   settings in `/etc/lvm/lvm.conf` to avoid slowdowns when interacting with
-   LXD.
-
-## Storage pool configuration
+{abbr}`LVM (Logical Volume Manager)` is a storage management framework rather than a file system.
+It is used to manage physical storage devices, allowing you to create a number of logical storage volumes that use and virtualize the underlying physical storage devices.
+
+Note that it is possible to over-commit the physical storage in the process, to allow flexibility for scenarios where not all available storage is in use at the same time.
+
+## Terminology
+
+LVM can combine several physical storage devices into a *volume group*.
+You can then allocate *logical volumes* of different types from this volume group.
+
+One supported volume type is a *thin pool*, which allows over-committing the resources by creating volumes that are larger than the available physical storage.
+Another type is a *volume snapshot*, which captures a specific state of a logical volume.
```

Comment:
I'm not sure if you want to go into it at this stage, but volume snapshots of thin vs non-thin logical volumes behave quite differently. 

For snapshots of thin logical volumes, you don't have to specify a size of the snapshot, and they don't consume allocated space until writes happen on the source logical volume. 

However with non-thin snapshots one has to specify a maximum size of the snapshot at creation time, this size is then taken away from the available space in the volume group. Note, this size is used to allow the volume snapshot to accommodate recording the differences as writes made to the source volume. 

This means that if a snapshot size is too small, then at some point in the future, depending on how many changes are made to the source volume, the snapshot can become "invalidated" and no longer usable. As you can image, as snapshots can be used as a form of "backup", this sort of behaviour is undesirable. So when LXD creates non-thin snapshots it uses the maximum size in order to accommodate the entire source volume from being re-written. This turns out to be essential with VMs because LVM non-thin doesn't support restoring a snapshot and leaving it the snapshot intact (unlike LVM thin volumes, BTRFS and ZFS). So in order to not destroy the snapshot when it is restored, we actually have to manually copy the contents of the snapshot over the top of the original volume (i.e we perform writes to the entire source volume). If we were not to set the snapshot size to the maximum size then restoring a snaphot would invalidate all the other snapshots (including itself).

So long story short, for LVM non-thin snaphots, they effectively take up the space of the source volume at create time.

See https://github.com/lxc/lxd/blob/master/lxd/storage/drivers/driver_lvm_volumes.go#L1203-L1224

Reply:
Ah I see you've covered some of this here https://github.com/lxc/lxd/pull/10643/files#diff-07ed16fddda9ff1e8130b444e649bd1086ece89e46c3c353ff66270dccc97271R25-R28

Maybe just add a mention below that as well as performance reductions, there is also space reduction.